### DISCUSSION TITLE
Request Reply Serialization

### DISCUSSION THREAD

**Discussion by alberk8** [nats-io/nats.net#484] [Q&A]
I have a remote App (servicing as Reply).   The issue is with parameter (AddEndPointAsync) of string or byte array where the Data is always null,   int and bool are ok.  Is this the way it is supposed to work?  

note: Nats .Net version 2.2.1
```cs
var natsOpt = new NatsOpts 
{
   SerializerRegistry = NatsJsonSerializerRegistry.Default,
   Url = "nats://192.168.100.2:4222",
}
var  _natsConnection = new NatsConnection(natsOpt);
var  _dataService = await _svc.AddServiceAsync("data", "1.0.0");
var  _group = await _dataService.AddGroupAsync("dataGroup, "dataQ");

_group.AddEndpointAsync<string>(GetDataAsync, "GetData");


private async ValueTask GetDataAsync(NatsSvcMsg<string> data)
{
   Console.WriteLine("Display: " +  data.Data)
   await data.ReplyAsync();
}
```

**Comment by mtmk** [comment]
You should use the default serializer for that. JSON serializers are for custom types. also have a look at the serialization docs https://nats-io.github.io/nats.net.v2/documentation/serialization.html

please let me know how it goes, if it's not clear etc. ;)

**Comment by alberk8** [comment]
Thank you.
